### ISSUE TITLE
Configure the existing Jetpack Remote Install flow

### ISSUE THREAD

**Issue by dvdchr** [wordpress-mobile/WordPress-iOS#20018]
Depends on #20017 

Reuse the existing Jetpack Remote Install flow to use the new plugin installation endpoint, skip the Jetpack Connection step, and update some copy for the UI.

Installation | Error
-|-
![jetpack_install_flow](https://user-images.githubusercontent.com/1299411/216925348-fdf61229-50c1-4b1f-923b-44b71232f55f.png) | ![error](https://user-images.githubusercontent.com/1299411/215046340-b50fa203-bd75-4b45-9a36-5504d3dc1d0a.png)

## Tasks

- [x] #20066 
- [x] #20108
- [x] #20130 
- [x] #20131
- [ ] #20143

## Todo
- [x] Install screen
  - [x] Performs request with the new endpoint when "Continue" is tapped (in #20130)
- [x] Loading screen
  - [x] Cancel the API request when the "Cancel" button is tapped (in #20143)
- [x] Complete screen
  - [x] Remove the Cancel button (in #20131)
  - [x] Update the subtitle text (in #20130)
  - [x] Update the primary button text to "Done" (in #20130)
- [x] Error screen
  - [x] Performs request to the new endpoint when "Retry" is tapped (in #20130)
  - [x] Contact Support button tags `origin:jp-install-full-plugin-error` as metadata (in #20143)
- [x] Adjust Tracks (in #20131)

### Others

- [x] Rename `ViewData` to `ViewModel` based on [this review comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/20066#discussion_r1104889541). (in #20143)

**Comment by dvdchr** [comment]
Updated issue description with the latest version of the design.

**Comment by dvdchr** [comment]
Some tasks will be moved to a separate issue for wiring:

- Dismiss the modal after "Cancel" button is tapped.
- Dismiss the modal after "Done" button is tapped.
- If needed, update the site data with the latest plugin info.
